### PR TITLE
Reorganize the config reference

### LINKED ISSUE #11253: Split out Teleport config yaml reference
## Details
The Teleport config YAML reference is quite long now (12 pages on a 4k monitor).  Having a tabbed format showing shorter examples and possibly still the long version would help so users can more easily navigate. 

Possible examples:

Simple configs:
- Proxy
- Auth
- Resource (ssh, db, windows desktop..)  Services

Advanced examples of:
- Proxy
- Auth
- other applicable examples

In the case of Teleport Cloud Only show resource examples. 

### Category
- Improve Existing

### PR DESCRIPTION
**Pull request by ptgott**
Closes #11253

This change breaks up the config reference by moving each Teleport service's configuration settings into a separate YAML snippet, and organizing each YAML snippet into a `TabItem` within a `Tabs` with `dropdownView` enabled. When Teleport Enterprise Cloud doesn't support a service, this change adds a `Notice` with a message about Teleport Enterprise Cloud in the relevant `TabItem`.

I thought this approach would be cleaner than alternatives:

- **Including an H2 for each service:** With this approach, the configuration reference would be even longer than it is now. With the approach I'm proposing, readers would navigate to "Reference configurations", then choose a service from a dropdown menu: much less scrolling for slightly more clicking.

- **Including another set of `TabItem`s for Teleport editions:** I thought about adding a `Tabs` with a dropdown option for each Teleport edition and a horizontal `TabItem` for each service, but this would require horizontal scrolling and doesn't scale very well.

- **Including horizontal tabs for each Teleport edition**, with an inner dropdown menu for each Teleport service: There is currently no way to do this without multiple `Tabs` components, which gets messy quickly.

**Note** that this change does not attempt to copy-edit comments/example values within the configuration snippets. The scope of this change is limited only to reorganization.

### CODE REVIEW

File: docs/pages/reference/config.mdx [hunk 1]
```diff
@@ -78,764 +78,105 @@ proxy_service:
...
-        # redirect:
-        #   - "localhost"
-        #   - "jenkins.internal.dev"
-        ## Optional list of extra headers to inject in to requests.
-        # headers:
-        #   For example:
-        #   - "Host: jenkins.example.com"
-
-## This section configures the 'kubernetes service'
 (!docs/pages/includes/kubernetes-access/kubernetes-config.yaml!)
```
Comment (alexfornuto) [inline docs/pages/reference/config.mdx:140]: Should we standardize the location of all these reference files?

Comment (alexfornuto) [reply docs/pages/reference/config.mdx:140]: And in that file, should we split the `command` array to be easier to read?

**Review by alexfornuto — APPROVED** [review]
This could go out as is, but I left some comments for possible improvements.
Additionally, I noticed that the indenting is not consistent between these example files, some use 2 spaces and others use 4.

File: docs/pages/includes/config-reference/instance-wide.yaml [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+    # Optional CA pin of the auth server. This enables a more secure way of
+    # adding new nodes to a cluster. See "Adding Nodes to your Cluster"
+    # (https://goteleport.com/docs/setup/admin/adding-nodes).
+    ca_pin: "sha256:7e12c17c20d9cb504bbcb3f0236be3f446861f1396dcbb44425fe28ec1c108f1"
+
+    # When running in multi-homed or NATed environments Teleport Nodes need
+    # to know which IP it will be reachable at by other Nodes.
+    #
+    # This value can be specified as FQDN e.g. host.example.com
+    advertise_ip: 10.1.0.5
```
[the remark this answers is not in the document]
Comment (ptgott) [reply docs/pages/includes/config-reference/instance-wide.yaml:67]: We should probably address this throughout the docs. I've created an issue: https://github.com/gravitational/teleport/issues/21778

File: docs/pages/includes/config-reference/instance-wide.yaml [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+
+
+    # Teleport provides HTTP endpoints for monitoring purposes. They are
+    # disabled by default but you can enable them using the diagnosis address.
+    # See the Teleport metrics reference:
+    # https://goteleport.com/docs/setup/diagnostics/metrics
```
Comment (alexfornuto) [inline docs/pages/includes/config-reference/instance-wide.yaml:73]: I just noticed while using our reference guide that this link is broken.

Comment (alexfornuto) [inline docs/pages/includes/config-reference/instance-wide.yaml:73]: I just noticed while using our reference guide that this link is broken.

Comment (alexfornuto) [reply docs/pages/includes/config-reference/instance-wide.yaml:73]:
```suggestion
    # https://goteleport.com/docs/management/diagnostics/metrics/
```

Comment (alexfornuto) [reply docs/pages/includes/config-reference/instance-wide.yaml:73]: We may also want to add a redirect for folks who already have this link copied into a local config file.

**Comment by zmb3** [comment]
I'm not sure this is in line with the goals set out in #15061.

I'd much prefer all the text be on one page where I can quickly cmd+f and search for it, rather than having to click through tabs.

Thoughts?

**Comment by alexfornuto** [comment]
searching through one long page works when you know what you're looking for. For those looking to learn about configs, separating them makes it easier to parse.

Additionally this page, as it is, is too big for our crawler:

> Record extracted is too big (108KB), the maximum is 97.65625KB

So: What about having instance-wide settings on this page, and sub-pages for each service type? That way each page could still be easily searchable when you know what service type it belongs to, and also easier for someone to learn from going top-to-bottom?

**Review by zmb3 — APPROVED** [review]

**Comment by ptgott** [comment]
@alexfornuto I pushed a change to split up the config reference using headings per Zac's feedback. I didn't want to split this into sub-pages, since the docs already has pages devoted to the configuration sections for specific Teleport services, and it would be pretty confusing to have, say, the Application Service config reference in the "Application Access" section (https://goteleport.com/docs/application-access/reference/) as well as in the Reference section.

**Review by alexfornuto — APPROVED** [review]

**Pull request merged**